src/GameLogic/Entities/Iceblock.ts

```diff
@@ -19,21 +19,25 @@ export class Iceblock implements Entity {
 
 	public direction: Direction8;
 
-	public containedEntity: Entity;
+	public containedEntity: Entity | undefined;
 
 	public melting: boolean;
 
 	public justPushed: boolean;
 
-	constructor(containedEntity: Entity) {
+	constructor(containedEntity: Entity | undefined) {
 		this.type = EntityType.Iceblock;
 		this.x = 0;
 		this.y = 0;
 		this.prevX = 0;
 		this.prevY = 0;
 		this.direction = Direction8.None;
 		this.containedEntity = containedEntity;
-		this.melting = containedEntity.type == EntityType.Fireball;
+		if (containedEntity === undefined) {
```

Comment:
I'd personally write it like this, but that's up to you, it's fine the way it is:
```
this.melting = containedEntity && containedEntity.type === EntityType.Fireball;
```